proto/podium/api/v1/podium.proto

```diff
@@ -0,0 +1,371 @@
+  int64 int_score = 3;
+
+  int32 rank = 4;
+  int32 previous_rank = 5;
+  int32 expire_at = 6;
+  int32 position = 7;
+}
+
+// ScoreChange is the score payload when upserting a score.
+message ScoreChange {
+  int64 score = 1;
+}
+
+message UpsertScoreRequest {
+  string leaderboard_id = 1;
+  string member_public_id = 2;
+  bool prev_rank = 3;
+  int32 scoreTTL = 4;
+  ScoreChange score_change = 5;
+}
+
+message TotalMembersRequest {
+  string leaderboard_id = 1;
+}
+
+message TotalMembersResponse {
+  bool success = 1;
+  int32 count = 2;
+}
+
+message IncrementScoreRequest {
+  string leaderboard_id = 1;
+  string member_public_id = 2;
+  int32 scoreTTL = 3;
+
+  // Body represents the increment payload.
+  message Body {
+    int64 increment = 1;
+  }
+  Body body = 4;
+}
+
+message GetMemberRequest {
+  string leaderboard_id = 1;
+  string member_public_id = 2;
+  string order = 3;
+  bool scoreTTL = 4;
+}
+
+message DefaultMemberResponse {
+  bool success = 1;
+  string publicID = 2;
+
+  //implemented as double because int64 is exported as string through grpc-gateway
+  double score = 3;
+
+  //the full int64 score received from Redis
+  int64 int_score = 4;
+  int32 rank = 5;
+  int32 previous_rank = 6;
+  int32 expire_at = 7;
+}
+
+message GetMembersRequest {
+  string leaderboard_id = 1;
+  string order = 2;
+  bool scoreTTL = 3;
+  string ids = 4;
+}
+
+message GetMembersResponse {
+  bool success = 1;
+  repeated MemberResponse members = 2;
+  repeated string notFound = 3;
+}
+
+message RemoveMemberRequest {
+  string leaderboard_id = 1;
+  string member_public_id = 2;
+}
+
+message RemoveMembersRequest {
+  string leaderboard_id = 1;
+  string ids = 2;
+}
+
+message BasicResponse {
+  bool success = 1;
+  string reason = 2;
+}
+
+message GetRankRequest {
+  string leaderboard_id = 1;
+  string member_public_id = 2;
+  string order = 3;
+}
+
+message GetRankResponse {
+  bool success = 1;
+  string publicID = 2;
+  int32 rank = 3;
+}
+
+message GetAroundMemberRequest {
+  string leaderboard_id = 1;
+  string member_public_id = 2;
+  string order = 3;
+  bool get_last_if_not_found = 4;
+  int32 page_size = 5;
+}
+
+message GetTopMembersRequest {
+  string leaderboard_id = 1;
+  int32 page_number = 2;
+  string order = 3;
+  int32 page_size = 5;
+}
+
+message GetTopPercentageRequest {
+  string leaderboard_id = 1;
+  int32 percentage = 2;
+  string order = 3;
+}
+
+message MultiUpsertScoreRequest {
+  string member_public_id = 1;
+  int32 scoreTTL = 2;
+  bool prev_rank = 3;
+
+  // ScoreMultiChange is the payload to update the score of a member on multiple leaderboards.
+  message ScoreMultiChange {
+    int64 score = 1;
+    repeated string leaderboards = 2;
+  }
+  ScoreMultiChange score_multi_change = 4;
+}
+
+message MultiUpsertScoreResponse {
+  bool success = 1;
+
+  // Member represents the information regarding a single member in response to a multi upsert score.
+  message Member {
```

Comment:
Could we make Member a top-level message and reuse where appropriate? Are there other entities that we could refactor out of messages as well?

Consider moving Member to its own file; as an entity it may benefit from having a place independent from the RPC service.

Reply:
Nice idea. I added a TODO to attack this at a latter point.